### ISSUE TITLE
[BUG] Hue & Saturation mode transition failed

### ISSUE THREAD

**Issue by xshuqun** [project-chip/connectedhomeip#36772]
### Reproduction steps

Facing a problem with Adaptive Lighting transiting to another color option

Some information about DUT:

1. **DUT is using FreeRTOS and LwIP.**
2. Thread priority: CHIP is 2 and LwIP is 9.
3. Messages received at through LwIP and has higher priority, so Postevent will take place.

After investigating the SDK, I observed the following:

1. The DUT is set to Adaptive Lighting by selecting it on the Homekit GUI.
4. Homekit sends a CommandRequest for `moveToColorTemp` and DUT handles callback `updateTempCommand` to set color temp.
5. Once the transition stabilizes, the DUT continuously triggers `updateTempCommand`.
6. A HueSaturation color option is selected on the Homekit GUI.
7. Homekit sends a CommandRequest `moveToHueAndSaturation` and DUT receives it + register timer with scheduleTimerCallbackMs for `updateHueSatCommand`.
8. Some Messages (for ReadRequest or other request) are received, which triggers PostEvent and added to the task queue (duplicate message packets are also observed).
9. The DUT enters RunEventLoop and continuously processes the task queue.
10. Homekit sends a CommandRequest to `stopAllColorTransitions` (sent by Apple on time, regardless of whether the DUT has completed the transition, I assumed).
11. The DUT handles `stopAllColorTransitions` and cancels/stops all ongoing timers for color control.
12. The DUT fails to complete the color transition.

Basically, before the DUT is able to handle the active timer, Homekit  has sent another CommandRequest to stop all color transition and cancel all the timers.

Expected outcome:

1. The DUT is set to Adaptive Lighting by selecting it on the Homekit GUI.
2. Homekit sends a CommandRequest for `moveToColorTemp` and DUT handles callback `updateTempCommand` to set color temp.
3. A HueSaturation color option is selected on the Homekit GUI.
4. Homekit sends a CommandRequest `moveToHueAndSaturation` and DUT receives it + register timer with scheduleTimerCallbackMs for `updateHueSatCommand`.
5. DUT triggers callback `updateHueSatCommand` and change Hue + Saturation.

I have attached the DUT with analysis of how FreeRTOS queue is being handled.
[1] [DUT_adaptive_lighting_success_logs.txt](https://github.com/user-attachments/files/18070738/DUT_adaptive_lighting_success_logs.txt)
[2] [DUT_adaptive_lighting_failed_logs.txt](https://github.com/user-attachments/files/18070737/DUT_adaptive_lighting_failed_logs.txt)

This is a slight video on the operation using Homekit:
[1] https://github.com/user-attachments/assets/9edbafef-4ed0-4b2d-9e3c-dfbea94c19ca

### Bug prevalence

always

### GitHub hash of the SDK that was being used

master

### Platform

ameba

**Comment by bzbarsky-apple** [comment]
@xshuqun So are you claiming this is a problem with the _server_ (and if so, what exactly is the problem), or with the _client_ (HomeKit) sending the wrong commands?

**Comment by xshuqun** [comment]
Apparently, I think the server hit a corner case where there is insufficient time to execute the necessary command.
However, whether the client is sending these command correctly, I am uncertain.

I had provided details in the description, where server (using FreeRTOS) has to clear its queue before handling active timer, but before the server could handle the timer, it was cancelled by the next command sent by the client. 

I would like some advise if this can be handled in another way or other solutions? (e.g., Like handling the active timer first and clearing the queue?)

**Comment by bzbarsky-apple** [comment]
@xshuqun Honestly, to me it sounds like the client is doing something wrong (stopping transitions before they have finished) and that you should report this issue to Apple....

**Comment by xshuqun** [comment]
I did report this issue to the Apple Team through the Feedback Assistant, but apparently, there is no response.

**Comment by bzbarsky-apple** [comment]
Yeah, that one is resolved with "ask the reporter these questions" back in October....  I have no idea why no one reached out to you, but I will try to prod it along.  Chances are nothing will happen until Jan 6 at the earliest, though; everyone involved is on vacation until then.

**Comment by xshuqun** [comment]
@bzbarsky-apple wondering if there is any feedback from the team on this discussion.

**Comment by woody-apple** [comment]
Please ping the feedback ticket at Apple, a CSA forum is not the place to provide feedback to Apple Home’s implementation
